compiler/src/dmd/parse.d

```diff
@@ -1298,12 +1298,38 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
             return 0;
         }
 
+        AST.Expression templateArgToExp(RootObject o, const ref Loc loc)
```

Comment:
I suggest going all in and making what follows the `@` a template argument and treat it like one. I.e. a RootObject rather than an Expression.

Reply:
Does that have any functional impact? It can be done, but currently `UserAttributeDeclaration` has a field `Expressions* atts` which is referenced in many different parts of the compiler, so it would require refactoring. I'd rather keep this PR about the syntax and not refactor the semantic code in it as well.

Reply:
@WalterBright

Reply:
It's already mostly there already, and wouldn't it be peculiar to be sort of half way in between expressions and template arguments?

Reply:
It's like that already. Note that this is purely a syntactical/parser change, no new semantics are introduced.

Reply:
I propose that the syntax not be changed, that it parses a template argument list, and any restrictions be imposed by the semantic pass.